frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js

```diff
@@ -1,6 +1,34 @@
 import React, { Component } from 'react';
+import gql from 'graphql-tag';
+import { graphql } from 'react-apollo';
 
 class CrossingDetails extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: props.crossing.name,
+      description: props.crossing.description
+    };
+  }
+
+  updateCrossing(e) {
+    debugger;
+    this.props.updateCrossingMutation({
+      variables: {
+        crossingId: this.props.crossing.id,
+        name: this.state.name,
+        description: this.state.description
+      }
+    })
+    .then(({ data }) => {
+      alert(data);
+    }).catch((error) => {
+      console.log('there was an error sending the query', error);
+    });
+  }
+
+  nameChanged = (e) => { this.setState({ name: e.target.value }) };
```

Comment:
Not entirely sure how React works, but do these onchange handlers need to be throttled at all? Not sure if React does this out of the box, or if a new update and render cycle based on a state change is called every time a type-y type-y happens.

Reply:
I haven't heard about this pattern being an issue in my research. @mateoclarke - any insights?

Reply:
Yeah, I think you'd need to manually throttle. I googled for a sec a didn't find a clear pattern either. Good thing to be aware but maybe the performance cost isn't that significant?